### PR TITLE
fixed crash in case there is no weather data

### PR DESCRIPTION
**Pull request by andweber**
The weather plugin should not crash in case there can be no data retrieved (see #475)

Simply added some checks if there is any data received and added an error msg for the user in case of no data. Please note: thats not a fix for #475. Thats still open.   

I also changed the testcase success criteria so that it does not fail in case that there is no data received. This might not be the intention and may need to be changed back.

### CODE REVIEW

File: plugins/speechhandler/weather/weather.py [hunk 1]
```diff
@@ -80,15 +80,23 @@ def get_weather(location, unit="f"):
...
+                           'env': 'store://datatables.org/alltableswithkeys'},
+                   headers={'User-Agent': 'Mozilla/5.0'})
     content = r.json()
-    channel = content['query']['results']['weather']['rss']['channel']
+    # make sure we got data
+    try:
+        channel = content['query']['results']['weather']['rss']['channel']
+    except:
+        # return empty Weather
+        return Weather(city=None,
```
Comment (Holzhaus) [inline plugins/speechhandler/weather/weather.py]: Let's just return `None` instead.

File: plugins/speechhandler/weather/weather.py [hunk 2]
```diff
@@ -80,15 +80,23 @@ def get_weather(location, unit="f"):
...
+                   params={'q': yql_query,
+                           'format': 'json',
+                           'env': 'store://datatables.org/alltableswithkeys'},
+                   headers={'User-Agent': 'Mozilla/5.0'})
     content = r.json()
-    channel = content['query']['results']['weather']['rss']['channel']
+    # make sure we got data
+    try:
+        channel = content['query']['results']['weather']['rss']['channel']
+    except:
```
Comment (Holzhaus) [inline plugins/speechhandler/weather/weather.py]: This should be `except KeyError:`.

File: plugins/speechhandler/weather/weather.py [hunk 3]
```diff
@@ -80,15 +80,23 @@ def get_weather(location, unit="f"):
...
+        # return empty Weather
+        return Weather(city=None,
+                       date=None,
+                       text=None,
+                       temp=None,
+                       forecast=None)
     current_date = dateutil.parser.parse(
-        channel['item']['condition']['date']).date()
+            channel['item']['condition']['date']).date()
     forecast = []
```
Comment (Holzhaus) [inline plugins/speechhandler/weather/weather.py]: Could you undo these whitespace changes since they don't belong into this commit.

**Comment by Holzhaus** [comment]
Thanks for contributing. I left you some minor nits. I'll merge as soon as these have been fixed.

**Comment by GeoBeBee** [comment]
The outstanding E402 would in fact be fixed by [#486](https://github.com/jasperproject/jasper-client/pull/486), ie. append '  # NOQA' to 4 lines.
Although it then has a problem with the weather which might be fixed by [#483](https://github.com/jasperproject/jasper-client/pull/483).

**Comment by GeoBeBee** [comment]
This Pull Request [#483](https://github.com/jasperproject/jasper-client/pull/483) is now covered by the successfully submitted [#493](https://github.com/jasperproject/jasper-client/pull/493) ('Default 4 syllable keyword HelloJasper works much better for new users').

**Pull request merged**